Vp2440 fixes

filipleple opened the pull request:
*ref: prot-1838*

mkopec committed:
build.sh: protectli: add default logo

Add the custom Protectli logo to built binaries, replacing the black
screen in the normal boot path and preserving it in recovery.

Upstream-Status: Inappropriate [Dasharo downstream]
Change-Id: I3ddbb41233403b89d2c8d823f0ae36abfaef8f0a
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.protectli_vp2430: bump version to v0.9.0-rc6

Upstream-Status: Inappropriate [Dasharo downstream]
Change-Id: Ia5bc9fbf711c02e1e038a7be5d646dc1e5f3e66e
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

filipleple committed:
configs/config.protectli_vp2430: Disable 1G pages and bump to rc7

ESXi’s bootloader doesn’t properly parse 1GB page entries when it
tries to replicate or interpret UEFI’s memory map/page tables during boot.
This fixes the issue of the ESXi installer crashing on startup.

Upstream-Status: Inappropriate [Dasharo downstream]
Change-Id: Icb8865da7b658d55e80190fa041abcf7876f24c5
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

mkopec committed:
configs/config.protectli_vp2430: bump version to v0.9.0

Upstream-Status: Inappropriate [Dasharo downstream]
Change-Id: I695d20de7a6215e31c2e13cc6cc54305d9d0e9c8
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

filipleple committed:
configs/config.protectli_vp2440: remove Intel ME options

Protectli don't want to include this feature in this release yet.

Upstream-Status: Inappropriate [Dasharo downstream]
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
configs/config.protectli_v*: use the black empty logo

Add blank default logo to payload to allow client customization without
showing Protectli branding. The logo is to be set afterwards, resulting
in the custom logo displaying on normal boot and the black one on
recovery.

This is adapted commit from dasharo-24.02.01 branch:
91140ddcc1c1

Upstream-Status: Inappropriate [Dasharo downstream]
Change-Id: I01b4c28a4f03b58c0bc856b03b645c45fa67c6af
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
mainboard/protectli/vault_adl_n: disable unused devices and params

Disables some devices and functionalities via devicetree and FSP params,
since they're unused and may impact boot time and performance.

Upstream-Status: Pending
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple commented:
@SergiiDmytruk we would need your help working with this commit: https://github.com/Dasharo/coreboot/pull/735/commits/27da76a9d9b0ef1ef2bae1d72f077e02613731ca

The problem is that on AMI firmware, all NICs have Wake-on-Lan enabled by default in Ubuntu. On coreboot, `enp2s0` sitting on `02:00.0` has it disabled.

I've tried:
* explicitly setting the PME_EN bit in the NIC's PCI config space in coreboot, similarly how it was disabled [here](https://github.com/Dasharo/coreboot/blob/dasharo/src/mainboard/acer/aspire_vn7_572g/mainboard.c#L133)
* setting fsp params that seemed relevant, e.g. `PchPmWolEnableOverride = 1`
* adding ACPI code for the NIC, declaring a `_PRW` method, mapping it to GPE69
  * I chose 69 since in Intel docs for ADL-N I found something along the lines of `PCI_EXP_STS / PCI_EXP_EN - shared GPE 0x69
Applies to devices on PCIe Root Ports` - seems suited for PCIe wake signals like PME# or LAN_WAKE#
  * this added a firmware_node to the NIC: 
     ```
     $ readlink /sys/bus/pci/devices/0000:02:00.0/firmware_node
     ../../../LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/device:0a/device:0b
     ```

Despite all this, gpe69 is still marked as disabled by OS:
   ```
   $ grep . /sys/firmware/acpi/interrupts/gpe69
        0         disabled     unmasked
   ```
  and enp2s0 wake is still disabled:
  ```
  $ sudo ethtool enp2s0 | grep Wake
      Supports Wake-on: pumbg
      Wake-on: d
  ```

I've no idea how to get this to work, and whether my approach is even correct. Perhaps AMI has a large EDK2-level driver responsible for this. I've left this revision flashed on the lab VP2440, in case you wanted to take a look

SergiiDmytruk commented:
Not sure I can help much, I only reviewed/rebased such changes.

> * explicitly setting the PME_EN bit in the NIC's PCI config space in coreboot

It doesn't seem to be set though:
```
root@3mdeb:~# lspci -xxx -s 02:00.0
02:00.0 Ethernet controller: Intel Corporation Ethernet Controller I226-V (rev 04)
00: 86 80 5c 12 06 04 10 00 04 00 00 02 10 00 00 00
10: 00 00 40 80 00 00 00 00 00 00 00 00 00 00 60 80
20: 00 00 00 00 00 00 00 00 00 00 00 00 86 80 00 00
30: 00 00 50 80 40 00 00 00 00 00 00 00 ff 01 00 00
40: 01 50 23 c8 08 20 00 00 00 00 00 00 00 00 00 00
50: 05 70 80 01 00 00 00 00 00 00 00 00 00 00 00 00
60: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
70: 11 a0 04 80 03 00 00 00 03 20 00 00 00 00 00 00
80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
90: 00 00 00 00 00 00 00 00 00 00 00 00 ff ff ff ff
a0: 10 00 02 00 c2 8c 00 10 3f 28 10 00 12 58 41 00
b0: 42 00 12 10 00 00 00 00 00 00 00 00 00 00 00 00
c0: 00 00 00 00 1f 08 00 00 00 04 00 00 00 00 00 00
d0: 02 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```
Setting the value via `setpci -s02:00.0 44.w=2108` works, but doesn't change anything, maybe device needs to be reset.

> Perhaps AMI has a large EDK2-level driver responsible for this.

Have you tried comparing PCI registers and ACPI in case of AMI with that of coreboot?

filipleple committed:
mb/protectli/vault_adl_n/Kconfig: disable S3 sleep

S3 sleep doesn't currently work on VP2440. Disable it on the platform,
so that the faulty state is never attempted.

Upstream-Status: Pending
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
configs/config.protectli_vp2440: bump to rc4

Upstream-Status: Inappropriate [Dasharo downstream]
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
configs/config.protectli_vp2440: Disable 1G pages

ESXi’s bootloader doesn’t properly parse 1GB page entries when it
tries to replicate or interpret UEFI’s memory map/page tables during boot.
This fixes the issue of the ESXi installer crashing on startup.

Upstream-Status: Inappropriate [Dasharo downstream]
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

macpijan commented:
@filipleple do we need to fix  `Dasharo Checks / check-defconfigs (push) Failing after 16s ` ?

macpijan commented on src/mainboard/protectli/vault_adl_n/romstage_fsp_params.c:
```diff
@@ -38,4 +38,11 @@ void mainboard_memory_init_params(FSPM_UPD *memupd)
+	memupd->FspmConfig.VtdBaseAddress[3] = 0;
+	memupd->FspmConfig.TcssDma0En = 0;
+	memupd->FspmConfig.TcssItbtPcie0En = 0;
+	memupd->FspmConfig.TcssXdciEn = 0;
```
do we need those extra settings or just this last one?

filipleple replied:
@macpijan Neither of these seem to affect the platform much, only `memupd->FspmConfig.TcssXhciEn = 0;` is crucial to be removed.

IDK about the configs, the error comes from the dasharo branch not this PR, I'm not sure i'll make it today

macpijan replied:
Let's not add useless changes then

filipleple committed:
mb/protectli/vault_adl_n/romstage_fsp_params.c: remove unnecessary changes

Some of the changes to FSP params were introduced when attempting to
resolve the MultiPhaseSiInit stage hang. Remove them, as they don't add
any value after all.

Upstream-Status: Pending
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple committed:
configs/config.protectli_vp2440: bump version to v0.9.0

Upstream-Status: Inappropriate [Dasharo downstream]
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple commented:
@macpijan I've removed the unnecessary changes and bumped the version to v0.9.0 non-rc

filipleple committed:
3rdparty/dasharo-blobs: update to origin/main

Upstream-Status: Inappropriate [Dasharo downstream]
Signed-off-by: Filip Lewiński <filip.lewinski@3mdeb.com>

filipleple commented:
The V54 build fail must've been a random network error, since it builds fine locally from the branch. Merging
